addons/mail/models/mail_message.py

```diff
@@ -690,7 +690,26 @@ def fetch(self, field_names):
         self = self.sudo()
         return super().fetch(field_names)
 
+    def _get_prev_write_date_by_id(self):
+        prev_write_date_by_id = dict((message_id, {}) for message_id in self.ids)
+        for message in self:
+            prev_write_date_by_id[message.id] = message.write_date
+        return prev_write_date_by_id
+
+    def _write_date(self, write_date_by_id):
+        self.flush_model()
+        for message in self:
+            super(Message, message)._write({ "write_date": write_date_by_id[message.id] })
+
+    def _write(self, vals):
+        prev_write_date_by_id = self._get_prev_write_date_by_id()
+        res = super(Message, self)._write(vals)
+        if not any(key in vals for key in ['body', 'attachment_ids', 'partner_ids']):
+            self._write_date(prev_write_date_by_id)
+        return res
```

Comment:
This method is the one that generates the `UPDATE` queries in database.
Making it call `self._write_date()`, which will generate more updates is not smart at all.
Performance-wise, it's a disaster.

Reply:
This is indeed poor performance-wise. My 1st imp was to conditionally not write on `write_date`, which was done with a key in context `skip_write_date`. This is ideal for performance, but without making any changes to `models.py` this is hardly feasible...